tools/collect-dependencies.js

```diff
@@ -13,7 +13,13 @@ if (option !== '-o' || !filename) {
 console.log('Starting to collect dependencies');
 
 console.log('Invoking pnpm list...');
-const output = execSync('pnpm list -w -r --parseable --depth=2').toString('utf8');
+
+// Warning: the maxBuffer has been increased to 10 MB as a quick solution.
+// In the case we hit the limit of the buffer again, we shall rewrite the
+// script for a better way to handle it.
+const output = execSync('pnpm list -w -r --parseable --depth=2', {
```

Comment:
@JerryHue A better fix for this would be to not buffer at all, and use the `stdout` stream you get back from `spawn` to `pipe` the output to an writable file stream.  There's no point holding anything in memory, just redirect it to disk.

Reply:
But we need to process the input from `stdout` to get the name of the packages, as well as only have unique project names (some project names appear several times due to some places using different versions).